### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by amartini51**
Minor documentation style and correctness fixes

Resolves: rdar://97022479&77649895&86212313&79689446&93208475

### CODE REVIEW

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 1]
```diff
@@ -41,8 +41,9 @@ def Availability(bits):
 
 if Self == 'Float16':
     SelfDocComment = '''\
-/// A half-precision (16b), floating-point value type.'''
-
+/// A half-precision (16b), floating-point value type.
+///
+/// `Float16` is available on Apple silicon, and unavailable on Intel.'''
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPointTypes.swift.gyb:46]: Need a "when targeting macOS" here; it _is_ available on Intel/Linux and Windows.

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 2]
```diff
@@ -53,7 +54,9 @@ elif Self == 'Double':
 
 elif Self == 'Float80':
     SelfDocComment = '''\
-/// An extended-precision, floating-point value type.'''
+/// An extended-precision, floating-point value type.
+///
+/// `Float80` is available on Intel, and unavailable on Apple silicon.'''
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPointTypes.swift.gyb:59]: This one is even more complex, it's probably best to say "Only available on Intel when the target system uses 80-bit long double" or similar.

File: stdlib/public/core/KeyPath.swift [hunk 3]
```diff
@@ -200,6 +200,14 @@ public class PartialKeyPath<Root>: AnyKeyPath { }
 internal enum KeyPathKind { case readOnly, value, reference }
 
 /// A key path from a specific root type to a specific resulting value type.
+///
+/// The most common way to make an instance of this type
+/// is by using a key-path expression like `\SomeClass.someProperty`.
+/// For more information,
+/// see [Key-Path Expressions[keypath] in *[The Swift Programming Language][tspl]*.
```
Comment (benrimmington) [inline stdlib/public/core/KeyPath.swift:207]:
```suggestion
/// see [Key-Path Expressions][keypath] in *[The Swift Programming Language][tspl]*.
```

File: stdlib/public/core/Optional.swift [hunk 4]
```diff
@@ -10,8 +10,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// A type that represents either a wrapped value or `nil`, the absence of a
-/// value.
+/// A type that represents either a wrapped value the absence of a value.
```
Comment (benrimmington) [inline stdlib/public/core/Optional.swift:13]:
```suggestion
/// A type that represents either a wrapped value or the absence of a value.
```

File: stdlib/public/core/FloatingPointTypes.swift.gyb (hunk 2 above)
Comment (amartini51) [reply stdlib/public/core/FloatingPointTypes.swift.gyb:59]: Changed in 4d35aae50e8.

File: stdlib/public/core/FloatingPointTypes.swift.gyb (hunk 1 above)
Comment (amartini51) [reply stdlib/public/core/FloatingPointTypes.swift.gyb]: Changed in 4d35aae50e8.

**Review by stephentyrone — APPROVED** [review]

**Comment by amartini51** [comment]
@swift-ci Please smoke test.

**Review by ktoso — APPROVED** [review]
Group changes LGTM, thanks!

**Review by bjlanier — APPROVED** [review]

**Pull request merged**